daemon/src/engine/server/sv_ccmds.cpp

```diff
@@ -369,6 +385,48 @@ static void SV_Systeminfo_f()
 	Info_Print( Cvar_InfoString( CVAR_SYSTEMINFO, false ) );
 }
 
+class ListMapsCmd: public Cmd::StaticCmd
+{
+public:
+	ListMapsCmd():
+		StaticCmd("listmaps", Cmd::SYSTEM, "Lists all maps available to the server")
+	{}
+
+	void Run(const Cmd::Args&) const OVERRIDE
+	{
+		std::vector<std::string> maps;
+
+		FS::RefreshPaks();
+		std::error_code ignore;
+		for ( const auto& pak : FS::GetAvailablePaks() )
+		{
+			FS::PakPath::LoadPakPrefix(pak, "maps", ignore);
+		}
+
+		for ( const auto& pak : FS::PakPath::ListFiles("maps", ignore) )
+		{
+			if ( Str::IsSuffix(".bsp", pak) )
```

Comment:
You also need to verify the bsp filename == pakfile name sans "map-". We will not load mismatching maps so listing them here is pointless.